src/runtime/meter-flow.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2021 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActivityIframeView} from '../ui/activity-iframe-view';
+import {SubscriptionFlows} from '../api/subscriptions';
+import {ViewSubscriptionsResponse} from '../proto/api_messages';
+import {feUrl} from './services';
+
+export class MeterFlow {
+  /**
+   * @param {!./deps.DepsDef} deps
+   */
+  constructor(deps) {
+    /** @private @const {!./deps.DepsDef} */
+    this.deps_ = deps;
+
+    /** @private @const {!Window} */
+    this.win_ = deps.win();
+
+    /** @private @const {!../components/activities.ActivityPorts} */
+    this.activityPorts_ = deps.activities();
+
+    /** @private @const {!../components/dialog-manager.DialogManager} */
+    this.dialogManager_ = deps.dialogManager();
+
+    /** @private @const {!ActivityIframeView} */
+    this.activityIframeView_ = new ActivityIframeView(
+      this.win_,
+      this.activityPorts_,
+      feUrl('/meteriframe', {
+        publicationId: deps.pageConfig().getPublicationId(),
+        origin: location.origin,
+      }),
+      {},
+      /* shouldFadeBody */ false
+    );
+  }
+
+  /**
+   * Starts the meter flow.
+   * @return {!Promise}
+   */
+  start() {
+    this.deps_.callbacks().triggerFlowStarted(SubscriptionFlows.SHOW_METER);
+    this.activityIframeView_.onCancel(() => {
+      this.deps_.callbacks().triggerFlowCanceled(SubscriptionFlows.SHOW_METER);
```

Comment:
Could you add a test for this?
https://app.codecov.io/gh/subscriptions-project/swg-js/compare/2050/diff

Reply:
Reverted.